node/src/effect/requests.rs

```diff
@@ -360,36 +360,39 @@ impl Display for StateStoreRequest {
 
 /// Details of a request for a list of deploys to propose in a new block.
 #[derive(DataSize, Debug)]
-pub struct ListForInclusionRequest {
+pub struct ProtoBlockRequest {
     /// The instant for which the deploy is requested.
     pub(crate) current_instant: Timestamp,
     /// Set of deploy hashes of deploys that should be excluded in addition to the finalized ones.
-    pub(crate) past_deploys: HashSet<DeployHash>,
+    pub(crate) past_deploys: Vec<DeployHash>,
     /// The height of the next block to be finalized at the point the request was made.
     /// This is _only_ a way of expressing how many blocks have been finalized at the moment the
     /// request was made. Block Proposer uses this in order to determine if there might be any
     /// deploys that are neither in `past_deploys`, nor among the finalized deploys it knows of.
     pub(crate) next_finalized: u64,
+    /// Random bit with which to construct the `ProtoBlock` requested.
+    pub(crate) random_bit: bool,
```

Comment:
My understanding of the goal of this ticket was that ProtoBlock should have differentiated `deploys` and `transfers` - so @EdHastingsCasperLabs suggested that I move the construction of the `ProtoBlock` into the `BlockProposer`. I haven't looked at `CandidateBlock`, but I like the idea of moving `random_bit` there instead of exposing raw `deploys` as opposed to a `ProtoBlock` (as the name `BlockProposer` would suggest).

Reply:
Let's split the work for moving `random_bit` out of this PR, as after an initial pass it looks like it could be a chunk of work and I'd like to get this merged.